### PR TITLE
Add support for DPS payload

### PR DESCRIPTION
**Pull request by prmathur-microsoft**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

**Comment by prmathur-microsoft** [comment]
https://github.com/Azure/iot-sdks-internals/pull/86 for API review

**Comment by prmathur-microsoft** [comment]
Noticed a regression. I will fix it and send updates. Please review this PR for design/implementation in the mean while.

**Comment by prmathur-microsoft** [comment]
@CIPop , @jebrando, @davilu   please review

File: provisioning/transport/amqp/src/MessageApplicationPropertyNames.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ internal static class MessageApplicationPropertyNames
         public const string OperationType = Prefix + "operation-type";
         public const string OperationId = Prefix + "operation-id";
         public const string Status = Prefix + "status";
+        public const string StatusCode = "statusCode"; // Is this needed?
```
Comment (timtay-microsoft) [inline provisioning/transport/amqp/src/MessageApplicationPropertyNames.cs]: >StatusCode [](start = 28, length = 10)

is this needed?

Comment (prmathur-microsoft) [reply provisioning/transport/amqp/src/MessageApplicationPropertyNames.cs]: @alextolp  is this needed?

Comment (alextolp) [reply provisioning/transport/amqp/src/MessageApplicationPropertyNames.cs]: Doesn't look like it's used anywhere.

**Pull request merged**